modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -155,20 +156,33 @@ private String getVersion(Extension extension)
 
     private void declareConfigurations(DeclarationDescriptor declaration, Class<?> extensionType)
     {
-        Configurations configurations = extensionType.getAnnotation(Configurations.class);
-        if (configurations != null)
+        Class<?>[] configurations = getConfigurationClasses(extensionType);
+        if (ArrayUtils.isEmpty(configurations))
         {
-            for (Class<?> declaringClass : configurations.value())
-            {
-                declareConfiguration(declaration, declaringClass);
-            }
+            declareConfiguration(declaration, extensionType);
```

Comment:
bug: if @Operations points to the same class which is annoteted with @Extension then the validation is not applied